File: service-1/src/main/java/com/nodelexchange/pingpong/service_1/configuration/MessageServiceConfig.java
package com.nodelexchange.pingpong.service_1.configuration;

import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class MessageServiceConfig {

    public static final String PUBLISH_QUEUE = "service-2-message-queue";
    public static final String SUBSCRIBE_QUEUE = "service-1-message-queue";

    @Bean
    public Queue queue() {
        return new Queue(SUBSCRIBE_QUEUE);
    }
}

File: service-2/src/main/kotlin/com/nodalexchange/pingpong/service_2/configuration/MessageServiceConfig.kt
package com.nodalexchange.pingpong.service_2.configuration

import org.springframework.amqp.core.Queue
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class MessageServiceConfig {

    companion object {
        const val PUBLISH_QUEUE = "service-1-message-queue"
        const val SUBSCRIBE_QUEUE = "service-2-message-queue"
    }

    @Bean
    fun queue(): Queue {
        return Queue(SUBSCRIBE_QUEUE)
    }
}